### ISSUE TITLE
Graceful handling of color contrast check fails for oklch and oklab with `none`

### ISSUE THREAD

**Issue by pattonwebz** [dequelabs/axe-core#4894]
### Product

axe-core

### Product Version

4.10.3

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation
Color contrast checks return results for issues on the page, or partial results in event of an error with a specific color.

#### Actual
Color contrast checks return an error with no results.

#### How to Reproduce
A page with CSS like this produces the issue: `oklch(0.961073 0.000047911 none / 0.2)`

#### Additional context
It seems the source of this is ColorJS treats `none` as `NaN` in the version used in axe-core. They have a beta that resolves this in the v6 line that handles none and NaN properly so those colors don't fail.

Related issue:
#4269

**Comment by namansancheti** [comment]
@straker 
I see, thanks for the detailed walkthrough on the required changes, in any case 👍

**Comment by chutchins25** [comment]
Should have posted in this sooner but I picked this up this morning and have begun looking into it.
